### PR TITLE
[RFC] Move to Ubuntu

### PR DESCRIPTION
**Pull request by nathanchance**
@tpimh had suggested this here: https://github.com/ClangBuiltLinux/continuous-integration/issues/34#issuecomment-441559837

It shrinks the image slightly (263 MB -> 229 MB) and avoids building binaries from source during the packaging process. While a non-x86 host still could not run the image generated from this Dockerfile because the builds of Clang from apt.llvm.org are for x86 hosts only, we're one step closer because we use .deb packages for everything.

Verification run: https://travis-ci.com/nathanchance/continuous-integration/builds/92515400

I've labelled this as RFC because I'm not married to the idea of switching distro bases but it does have its benefits so it's worth discussing.

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
It seems this conflates moving to ubuntu with getting qemu from Joel's PPA? Is it possible to fetch qemu from Joel's PPA while still on Debian?

**Comment by nathanchance** [comment]
> It seems this conflates moving to ubuntu with getting qemu from Joel's PPA? Is it possible to fetch qemu from Joel's PPA while still on Debian?

We currently do: https://github.com/ClangBuiltLinux/continuous-integration/blob/master/debian-setup.sh#L32-L42

It's fragile though because of the dependencies; Ubuntu could update some dependency that Debian doesn't, which is what I want to get away from.

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+FROM ubuntu:rolling
 
 # Make sure that all packages are up to date then
-# install the base Debian packages that we need for
-# building the kernel and QEMU
+# install the base Ubuntu packages that we need for
+# building the kernel
 RUN apt-get update -qq && \
     apt-get upgrade -y && \
+    ln -snf /usr/share/zoneinfo/UTC /etc/localtime && echo "UTC" > /etc/timezone && \
```
Comment (nickdesaulniers) [inline Dockerfile]: why is this required?

Comment (nathanchance) [reply Dockerfile]: https://travis-ci.com/nathanchance/dockerimage/builds/92513642

There doesn't appear to be a default timezone.

**Comment by nickdesaulniers** [comment]
> Ubuntu could update some dependency that Debian doesn't, 

so the argument in favor of ubuntu is that they update packages *more frequently* thank debian?

> which is what I want to get away from.

but this PR moves *to* ubunutu.

File: Dockerfile (hunk 1 above)
Comment (nickdesaulniers) [reply Dockerfile]: haha interesting, thanks for the background

**Comment by nathanchance** [comment]
> > Ubuntu could update some dependency that Debian doesn't,
> 
> so the argument in favor of ubuntu is that they update packages _more frequently_ than debian?
> 
> > which is what I want to get away from.
> 
> but this PR moves _to_ ubunutu.

Sorry, I meant ultimately I want to avoid having a package built for Ubuntu installed on Debian. The commit in our CI repo that started that process explains it more: https://github.com/ClangBuiltLinux/continuous-integration/commit/b7ee15eb433625b5e616d8c91f85d266b47dfd9f

Relying on another distribution's packages to mostly match another's in order to install a package just feels fragile to me. Maybe I am being paranoid though.

**Comment by nickdesaulniers** [comment]
ah, maybe add that to a commit message: `move from debian to ubuntu because it's a smaller docker image and Joel's PPA is for Ubuntu, not Debian` or something.

**Comment by nathanchance** [comment]
> ah, maybe add that to a commit message: `move from debian to ubuntu because it's a smaller docker image and Joel's PPA is for Ubuntu, not Debian` or something.

Done.

> Please wait for @shenki 's review before merging.

Thank you, will do.
